setup.py

```diff
@@ -87,7 +87,8 @@ def build_extension(self, ext):
         # setuptools install
         f = open(os.path.join(extdir, 'teca.py'), 'w')
         f.write('import sys, os\n')
-        f.write('from teca import *\n')
+        f.write('sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)),"lib"))\n')
+        f.write('from teca_py import *\n')
```

Comment:
are you trying to run out of the build itself? I think that's what __file__ here will point to. I suspect that this would break the install?

Reply:
No, since `__file__` is within quotes here, it written to `teca.py` (and not resolved during the setup.py process, which I think is what you are implying), this will refer to the file path of `teca.py` once imported: e.g., as in `import teca; print(teca.__file__)`.  So it will refer to the installed path.  This is intended to add the `lib` subdirectory of the TECA install to the path, which is needed since `teca_py.py` imports things like `teca_tc_stats.py`, which exist in lib but aren't part of the path.

Reply:
OK, got it. I hadn't taken account of the quotes.  I vaguely recall that the solution as implemented was taken directly from the Python packaging documentation. I'll try to find it. Also, I'm very curious as to why all this works for me and not you.

Reply:
Perhaps we should have a working (virtual w/ screenshare?) session in the next week or so.

Reply:
Btw when you say 'this works', are you referring to the set of issues described in #213?

Reply:
Yup, I was thinking the same thing. A face to face would be the best way to proceed.

`python setup.py install` run from a clone has been working for me on Fedora 28, Ubuntu 18.04, and Mac OS latest. I had been using virtual env's to keep the installs out of the system places.

I believe `$ORIGIN` is a Linux specific trick, and while Mac OS has it's own mechanisms, it would be best to not go into platform specific tricks if we can avoid them.

I would like to, understand what is different about your workflow/environment and mine, and test this on Fedora, Ubuntu and Mac, and in PyPi before merging it.

https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling
http://man7.org/linux/man-pages/man8/ld.so.8.html
https://packaging.python.org/guides/packaging-namespace-packages/

Reply:
I see the issue now. What I had initially done never worked. my testing was not thorough enough, I had been only testing `from teca import *`, which does succeed but nothing is imported. At one point I had a solution similar to yours manipulating paths in the generated file, without using `RPATH`, but I backed it out after being able to import without it.

Reply:
okay, glad to confirm that we're now on the same page wrt the issue.